src/providers/FusionAuthProvider.tsx

```diff
@@ -66,19 +83,32 @@ async function generateUrl(
     functionType: FunctionType,
     baseUrl: string,
     clientID: string,
-    scope: string,
     redirectUri: string,
+    scope: string,
     state = '',
+    idTokenHint = '',
 ) {
-    const queryString = new URLSearchParams({
-        client_id: clientID,
-        scope: scope,
-        response_type: 'code',
-        redirect_uri: redirectUri,
-        code_challenge: await generatePKCE(),
-        code_challenge_method: 'S256',
-        state: `${generateRandomString()}:${state}`,
-    });
+    let queryString: URLSearchParams;
+    if (
+        functionType == FunctionType.login ||
+        functionType == FunctionType.register
+    ) {
+        queryString = new URLSearchParams({
+            client_id: clientID,
+            scope: scope,
+            response_type: 'code',
+            redirect_uri: redirectUri,
+            code_challenge: await generatePKCE(),
+            code_challenge_method: 'S256',
+            state: `${generateRandomString()}:${state}`,
+        });
+    } else {
+        queryString = new URLSearchParams({
+            client_id: clientID,
+            post_logout_redirect_uri: redirectUri,
+            id_token_hint: idTokenHint,
+        });
+    }
 
     return `${baseUrl}/${functionType}?${queryString}`;
```

Comment:
I think we should actually make this function more generic so that it takes an optional `object` that it will use for the query params, that way the call site determines what the URL params are entirely, and we don't need any conditional logic here based on function type